File: Backend/src/main/java/com/project/autoserve/service/AppointmentService.java
package com.project.autoserve.service;

import java.util.List;

import com.project.autoserve.dto.appointment.AppointmentRequestDTO;
import com.project.autoserve.dto.appointment.AppointmentResponseDTO;
import com.project.autoserve.enums.AppointmentStatus;

public interface AppointmentService {

    /**
     * Book a new appointment.
     *
     * @param request Appointment details.
     * @param userEmail Logged-in user's email.
     * @return Booked appointment.
     */
    AppointmentResponseDTO bookAppointment(
            AppointmentRequestDTO request,
            String userEmail);

    /**
     * Get all appointments of the logged-in user.
     *
     * @param userEmail Logged-in user's email.
     * @return List of appointments.
     */
    List<AppointmentResponseDTO> getMyAppointments(
            String userEmail,
            String search);

    /**
     * Get appointment by ID.
     *
     * @param appointmentId Appointment ID.
     * @return Appointment details.
     */
    AppointmentResponseDTO getAppointmentById(
            Long appointmentId);
    
    List<AppointmentResponseDTO> getAllAppointments();
    
    AppointmentResponseDTO assignMechanic(
            Long appointmentId,
            Long mechanicId
    );

    AppointmentResponseDTO updateAppointmentStatus(
            Long appointmentId,
            AppointmentStatus status
    );
    
    AppointmentResponseDTO cancelAppointment(
            Long appointmentId,
            String userEmail
    );

}

File: Frontend/src/services/appointmentService.js
import axiosInstance from "../api/axiosInstance";

export const appointmentService = {
  createAppointment: async (appointmentData) => {
    const response = await axiosInstance.post("/appointments", appointmentData);
    return response.data;
  },

  getAppointments: async (search = "") => {
    const response = await axiosInstance.get("/appointments", {
      params: { search },
    });
    return response.data;
  },

  getAllAppointments: async () => {
    const response = await axiosInstance.get("/appointments/all");
    return response.data;
  },

  getAppointmentById: async (id) => {
    const response = await axiosInstance.get(`/appointments/${id}`);
    return response.data;
  },

  updateAppointmentStatus: async (id, status) => {
    const response = await axiosInstance.put(`/appointments/${id}/status`, {
      status,
    });
    return response.data;
  },

  assignMechanic: async (appointmentId, mechanicId) => {
    const response = await axiosInstance.put(
      `/appointments/${appointmentId}/assign`,
      { mechanicId },
    );
    return response.data;
  },

  cancelAppointment: async (appointmentId) => {
    const response = await axiosInstance.put(
      `/appointments/${appointmentId}/cancel`,
    );

    return response.data;
  },
};
